db: lower bound violation in pebble tests

```
panic: lower bound f#12,21 violated by key f#14,21; wraps *keyspan.assertIter

goroutine 78074 [running]:
github.com/cockroachdb/pebble/internal/keyspan.(*assertIter).panicf(0xc00097dc20, {0x10f22786e?, 0x1?}, {0xc0008ac648?, 0x1?, 0x1?})
	/Users/runner/work/pebble/pebble/internal/keyspan/assert_iter.go:82 +0xa8
github.com/cockroachdb/pebble/internal/keyspan.(*assertIter).check(0xc00097dc20, 0xc0008074b0)
	/Users/runner/work/pebble/pebble/internal/keyspan/assert_iter.go:99 +0x1d4
github.com/cockroachdb/pebble/internal/keyspan.(*assertIter).First(0xc00097dc20)
	/Users/runner/work/pebble/pebble/internal/keyspan/assert_iter.go:134 +0x45
github.com/cockroachdb/pebble/internal/keyspan.(*filteringIter).First(0xc000807500)
	/Users/runner/work/pebble/pebble/internal/keyspan/filter.go:82 +0x23
github.com/cockroachdb/pebble/internal/keyspan.(*assertIter).First(0xc00097dcb0)
	/Users/runner/work/pebble/pebble/internal/keyspan/assert_iter.go:133 +0x23
github.com/cockroachdb/pebble/internal/keyspan.(*DefragmentingIter).First(0xc0002acb40)
	/Users/runner/work/pebble/pebble/internal/keyspan/defragment.go:289 +0x24
github.com/cockroachdb/pebble/internal/keyspan/keyspanimpl.(*MergingIter).First(0xc0003b9508)
	/Users/runner/work/pebble/pebble/internal/keyspan/keyspanimpl/merging_iter.go:615 +0x6a
github.com/cockroachdb/pebble.(*DB).loadTableRangeDelStats(0xc0008e4008, {0x10f5220b8, 0xc0009043c0}, 0xc0008f4c08, 0x6, 0xc0002856c0, 0xc0008acc08)
	/Users/runner/work/pebble/pebble/table_stats.go:375 +0xfb
github.com/cockroachdb/pebble.(*DB).loadTableStats.func1({0x10f5220b8, 0xc0009043c0})
	/Users/runner/work/pebble/pebble/table_stats.go:316 +0xf9
github.com/cockroachdb/pebble.(*tableCacheContainer).withCommonReader(0xc000072188?, 0xc0002856c0, 0xc0008acc50)
	/Users/runner/work/pebble/pebble/table_cache.go:295 +0xf3
github.com/cockroachdb/pebble.(*DB).loadTableStats(0x5?, 0x0?, 0x0?, 0x0?)
	/Users/runner/work/pebble/pebble/table_stats.go:305 +0xd4
github.com/cockroachdb/pebble.(*DB).scanReadStateTableStats(0xc0008e4008, 0xc0000b4d20, {0xc00019f490, 0x0, 0x32})
	/Users/runner/work/pebble/pebble/table_stats.go:281 +0x925
github.com/cockroachdb/pebble.(*DB).collectTableStats(0xc0008e4008)
	/Users/runner/work/pebble/pebble/table_stats.go:113 +0x236
created by github.com/cockroachdb/pebble.(*DB).maybeCollectTableStatsLocked in goroutine 78089
FAIL	github.com/cockroachdb/pebble	44.976s
```

https://github.com/cockroachdb/pebble/actions/runs/8102029032/job/22143601288?pr=3363

--- a/table_cache_test.go
+++ b/table_cache_test.go
@@ -263,13 +263,13 @@ func TestVirtualReadsWiring(t *testing.T) {
 	b := newBatch(d)
 	// Some combination of sets, range deletes, and range key sets/unsets, so
 	// all of the table cache iterator functions are utilized.
-	require.NoError(t, b.Set([]byte{'a'}, []byte{'a'}, nil))
-	require.NoError(t, b.Set([]byte{'d'}, []byte{'d'}, nil))
-	require.NoError(t, b.DeleteRange([]byte{'c'}, []byte{'e'}, nil))
-	require.NoError(t, b.Set([]byte{'f'}, []byte{'f'}, nil))
-	require.NoError(t, b.RangeKeySet([]byte{'f'}, []byte{'k'}, nil, []byte{'c'}, nil))
-	require.NoError(t, b.RangeKeyUnset([]byte{'j'}, []byte{'k'}, nil, nil))
-	require.NoError(t, b.Set([]byte{'z'}, []byte{'z'}, nil))
+	require.NoError(t, b.Set([]byte{'a'}, []byte{'a'}, nil))                           // SeqNum start.
+	require.NoError(t, b.Set([]byte{'d'}, []byte{'d'}, nil))                           // SeqNum: +1
+	require.NoError(t, b.DeleteRange([]byte{'c'}, []byte{'e'}, nil))                   // SeqNum: +2
+	require.NoError(t, b.Set([]byte{'f'}, []byte{'f'}, nil))                           // SeqNum: +3
+	require.NoError(t, b.RangeKeySet([]byte{'f'}, []byte{'k'}, nil, []byte{'c'}, nil)) // SeqNum: +4
+	require.NoError(t, b.RangeKeyUnset([]byte{'j'}, []byte{'k'}, nil, nil))            // SeqNum: +5
+	require.NoError(t, b.Set([]byte{'z'}, []byte{'z'}, nil))                           // SeqNum: +6
 	require.NoError(t, d.Apply(b, nil))
 	require.NoError(t, d.Flush())
 	require.NoError(t, d.Compact([]byte{'a'}, []byte{'b'}, false))
@@ -287,6 +287,13 @@ func TestVirtualReadsWiring(t *testing.T) {
 	f2 := f1 + 1
 	d.mu.versions.nextFileNum += 2
 
+	seqNumA := parentFile.Smallest.SeqNum()
+	// See SeqNum comments above.
+	seqNumCEDel := seqNumA + 2
+	seqNumRangeSet := seqNumA + 4
+	seqNumRangeUnset := seqNumA + 5
+	seqNumZ := seqNumA + 6
+
 	v1 := &manifest.FileMetadata{
 		FileBacking:      parentFile.FileBacking,
 		FileNum:          f1,
@@ -294,10 +301,10 @@ func TestVirtualReadsWiring(t *testing.T) {
 		Size:             parentFile.Size / 2,
 		SmallestSeqNum:   parentFile.SmallestSeqNum,
 		LargestSeqNum:    parentFile.LargestSeqNum,
-		Smallest:         base.MakeInternalKey([]byte{'a'}, parentFile.Smallest.SeqNum(), InternalKeyKindSet),
-		Largest:          base.MakeInternalKey([]byte{'a'}, parentFile.Smallest.SeqNum(), InternalKeyKindSet),
-		SmallestPointKey: base.MakeInternalKey([]byte{'a'}, parentFile.Smallest.SeqNum(), InternalKeyKindSet),
-		LargestPointKey:  base.MakeInternalKey([]byte{'a'}, parentFile.Smallest.SeqNum(), InternalKeyKindSet),
+		Smallest:         base.MakeInternalKey([]byte{'a'}, seqNumA, InternalKeyKindSet),
+		Largest:          base.MakeInternalKey([]byte{'a'}, seqNumA, InternalKeyKindSet),
+		SmallestPointKey: base.MakeInternalKey([]byte{'a'}, seqNumA, InternalKeyKindSet),
+		LargestPointKey:  base.MakeInternalKey([]byte{'a'}, seqNumA, InternalKeyKindSet),
 		HasPointKeys:     true,
 		Virtual:          true,
 	}
@@ -310,12 +317,12 @@ func TestVirtualReadsWiring(t *testing.T) {
 		Size:             parentFile.Size / 2,
 		SmallestSeqNum:   parentFile.SmallestSeqNum,
 		LargestSeqNum:    parentFile.LargestSeqNum,
-		Smallest:         base.MakeInternalKey([]byte{'d'}, parentFile.Smallest.SeqNum()+2, InternalKeyKindRangeDelete),
-		Largest:          base.MakeInternalKey([]byte{'z'}, parentFile.Largest.SeqNum(), InternalKeyKindSet),
-		SmallestPointKey: base.MakeInternalKey([]byte{'d'}, parentFile.Smallest.SeqNum()+2, InternalKeyKindRangeDelete),
-		LargestPointKey:  base.MakeInternalKey([]byte{'z'}, parentFile.Largest.SeqNum(), InternalKeyKindSet),
-		SmallestRangeKey: base.MakeInternalKey([]byte{'f'}, parentFile.Smallest.SeqNum()+2, InternalKeyKindRangeKeySet),
-		LargestRangeKey:  base.MakeInternalKey([]byte{'k'}, parentFile.Largest.SeqNum(), InternalKeyKindRangeKeyUnset),
+		Smallest:         base.MakeInternalKey([]byte{'d'}, seqNumCEDel, InternalKeyKindRangeDelete),
+		Largest:          base.MakeInternalKey([]byte{'z'}, seqNumZ, InternalKeyKindSet),
+		SmallestPointKey: base.MakeInternalKey([]byte{'d'}, seqNumCEDel, InternalKeyKindRangeDelete),
+		LargestPointKey:  base.MakeInternalKey([]byte{'z'}, seqNumZ, InternalKeyKindSet),
+		SmallestRangeKey: base.MakeInternalKey([]byte{'f'}, seqNumRangeSet, InternalKeyKindRangeKeySet),
+		LargestRangeKey:  base.MakeInternalKey([]byte{'k'}, seqNumRangeUnset, InternalKeyKindRangeKeyUnset),
 		HasPointKeys:     true,
 		Virtual:          true,
 	}

